vassal-app/src/main/java/VASSAL/counters/NonRectangular.java

```diff
@@ -142,28 +148,36 @@ private Shape buildPath(String spec) {
             break;
           case 'm':
             path.moveTo(Integer.parseInt(st.nextToken()),
-                        Integer.parseInt(st.nextToken()));
+              Integer.parseInt(st.nextToken()));
             break;
           case 'l':
             path.lineTo(Integer.parseInt(st.nextToken()),
-                        Integer.parseInt(st.nextToken()));
+              Integer.parseInt(st.nextToken()));
+            break;
+          // Note the image name is stored as a single token so older clients will ignore it.
+          case 'n':
+            imageName = token.length() > 1 ? token.substring(1) : "";
             break;
           }
         }
         sh = new Area(path);
-        shapeCache.put(spec, sh);
+        cache.put(spec, new ImmutablePair<>(imageName, sh));
```

Comment:
Tip: Pair.of() will make you an ImmutablePair and is shorter. Not a thing we need often; I only just discovered that.